include/fms_platform.h

```diff
@@ -92,8 +92,8 @@
 #define NF_GET_ATT_REAL nf_get_att_double
 #endif
 
-#if defined __CRAYXT_COMPUTE_LINUX_TARGET || defined __GFORTRAN__
-!Cray XT compilers do not support real*16 computation
+#if defined __CRAYXT_COMPUTE_LINUX_TARGET || defined __PGI
+!Cray XT and PGI compilers do not support real*16 computation
```

Comment:
The correct way to accomplish this is to set some macro like HAVE_QUAD_REAL in configure.ac, and use it everywhere in the code.

Then, we could take the next step, which is to write a configure check which determines whether the compiler supports real*8. Then this code will work on all compilers, even ones that we don't know about or test on.